libr/core/cmd_anal.c

```diff
@@ -125,13 +123,54 @@ static int var_cmd (RCore *core, const char *str) {
 		return false;
 	}
 	/* Variable access CFvs = set fun var */
-	if (str[0] == 'a' && r_config_get_i (core->config, "anal.vars")) {
-		r_anal_var_delete_all (core->anal, fcn->addr, 'r');
-		r_anal_var_delete_all (core->anal, fcn->addr, 'b');
-		r_anal_var_delete_all (core->anal, fcn->addr, 's');
-		fcn_callconv (core, fcn);
+	switch (str[0]) {
+	case 'a':
+		if (r_config_get_i (core->config, "anal.vars")) {
+			r_anal_var_delete_all (core->anal, fcn->addr, 'r');
+			r_anal_var_delete_all (core->anal, fcn->addr, 'b');
+			r_anal_var_delete_all (core->anal, fcn->addr, 's');
+			fcn_callconv (core, fcn);
+		}
+		return true;
+	case 'n': {
+		RAnalVar *v1;
+		char *str_dup = strdup (str);
+		char *old_name = r_str_trim_head (strchr (str_dup, ' '));
+		char *new_name = strchr (old_name, ' ');
+		if (!new_name) {
+			var_help (core, '?');
+			return false;
+		}
+		*new_name++ = 0;
+		r_str_chop (new_name);
+		v1 = r_anal_var_get_byname (core->anal, fcn, 'r', new_name);
+		if (!v1) {
+			v1 = r_anal_var_get_byname (core->anal, fcn, 'b', new_name);
+		}
+		if (!v1) {
+			v1 = r_anal_var_get_byname (core->anal, fcn, 's', new_name);
+		}
+		if(v1) {
+			r_anal_var_free (v1);
+			eprintf ("variable or arg with name `%s` already exist\n", new_name);
```

Comment:
this check must be done by r_anal_var_rename. otherwise we will have to duplicate the same checks on every var_rename call